tests/test_tree.rs

```diff
@@ -162,4 +159,128 @@ fn recovery() {
     t.__delete_all_files();
 }
 
-// TODO quickcheck splits, reads, writes interleaved
+#[derive(Debug, Clone)]
+enum Op {
+    Set(u8, u8),
+    Get(u8),
+    Del(u8),
+    Cas(u8, u8, u8),
+    Scan(u8, usize),
+    Restart,
+}
+
+impl Arbitrary for Op {
+    fn arbitrary<G: Gen>(g: &mut G) -> Op {
+        if g.gen_weighted_bool(50) {
+            return Op::Restart;
+        }
+
+        let choice = g.gen_range(0, 5);
+
+        match choice {
+            0 => Op::Set(g.gen::<u8>(), g.gen::<u8>()),
+            1 => Op::Get(g.gen::<u8>()),
+            2 => Op::Del(g.gen::<u8>()),
+            3 => Op::Cas(g.gen::<u8>(), g.gen::<u8>(), g.gen::<u8>()),
+            4 => Op::Scan(g.gen::<u8>(), g.gen_range::<usize>(0, 40)),
+            _ => panic!("impossible choice"),
```

Comment:
we generate a sequence of operations on the tree, hoping to trigger bugs. similar to fuzzing